src/NuGet.Services.BasicSearch/ServiceEndpoints.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Owin;
+using NuGet.Indexing;
+
+namespace NuGet.Services.BasicSearch
+{
+    public class ServiceEndpoints
+    {
+        public static async Task V3SearchAsync(IOwinContext context, NuGetSearcherManager searcherManager)
+        {
```

Comment:
Think we should have the same validation on the provided parameters inside NuGet.Indexing. See my earlier comment.